tests/unit/job/test_query.py

```diff
@@ -1037,6 +1037,85 @@ def test_result_dry_run(self):
         self.assertIsNone(result.job_id)
         self.assertIsNone(result.query_id)
 
+    # If the job doesn't exist, create the job first. Issue:
+    # https://github.com/googleapis/python-bigquery/issues/1940
+    def test_result_begin_job_if_not_exist(self):
+        begun_resource = self._make_resource()
+        query_running_resource = {
+            "jobComplete": True,
+            "jobReference": {
+                "projectId": self.PROJECT,
+                "jobId": self.JOB_ID,
+                "location": "US",
+            },
+            "schema": {"fields": [{"name": "col1", "type": "STRING"}]},
+            "status": {"state": "RUNNING"},
+        }
+        query_done_resource = {
+            "jobComplete": True,
+            "jobReference": {
+                "projectId": self.PROJECT,
+                "jobId": self.JOB_ID,
+                "location": "US",
+            },
+            "schema": {"fields": [{"name": "col1", "type": "STRING"}]},
+            "status": {"state": "DONE"},
+        }
+        done_resource = copy.deepcopy(begun_resource)
+        done_resource["status"] = {"state": "DONE"}
+        connection = make_connection(
+            begun_resource,
+            query_running_resource,
+            query_done_resource,
+            done_resource,
+        )
+        client = _make_client(project=self.PROJECT, connection=connection)
+        job = self._make_one(self.JOB_ID, self.QUERY, client)
+        job._properties["jobReference"]["location"] = "US"
+
+        job.result()
+
+        create_job_call = mock.call(
+            method="POST",
+            path=f"/projects/{self.PROJECT}/jobs",
+            data={
+                "jobReference": {
+                    "jobId": self.JOB_ID,
+                    "projectId": self.PROJECT,
+                    "location": "US",
+                },
+                "configuration": {
+                    "query": {"useLegacySql": False, "query": self.QUERY},
+                },
+            },
+            timeout=None,
+        )
+        reload_call = mock.call(
+            method="GET",
+            path=f"/projects/{self.PROJECT}/jobs/{self.JOB_ID}",
+            query_params={"projection": "full", "location": "US"},
+            timeout=DEFAULT_GET_JOB_TIMEOUT,
+        )
+        get_query_results_call = mock.call(
+            method="GET",
+            path=f"/projects/{self.PROJECT}/queries/{self.JOB_ID}",
+            query_params={
+                "maxResults": 0,
+                "location": "US",
+            },
+            timeout=None,
+        )
+
+        # assert connection.api_request.call_args == []
+        connection.api_request.assert_has_calls(
+            [
+                create_job_call,
```

Comment:
Since this is the most important assertion, let's add a comment so folks don't remove it by accident.

```suggestion
                # Make sure we start a job that hasn't started yet. See:
                # https://github.com/googleapis/python-bigquery/issues/1940
                create_job_call,
```